packages/relay/src/lib/eth.ts

```diff
@@ -1618,6 +1619,11 @@ export class EthImpl implements Eth {
       // depending on stage of contract execution revert the result.to value may be null
       if (!_.isNil(result.to)) {
         if(showDetails) {
+          // check the size of the block before querying for transaction details
+          if (blockResponse.count >= this.ethGetTransactionCountMaxBlockRange) {
```

Comment:
Not sure this is the right check. 
If we're just checking the blockSize then this could go much higher in this method to save on API calls and logic.
`blockResponse.count` is the number of transaction overall in the block, however, on Hedera that also includes non-EVM transactions.

What we should be doing is checking the number of EVM transaction (`contractResults`) to see if it's larger than `this.ethGetTransactionCountMaxBlockRange`.

Reply:
Done.